docker-compose.yml

```
@@ -0,0 +1,233 @@
+        else
+          /google-cloud-sdk/bin/dev_appserver.py /app/oppia/app_dev.yaml --runtime python38 --host=0.0.0.0 --port=8181 --admin_host=0.0.0.0 --admin_port=8000 --skip_sdk_update_check=True
+        fi
+      '
+
+  datastore:
+    build:
+      context: .
+      dockerfile: Dockerfile.google_cloud_sdk
+    container_name: oppia-cloud-datastore
+    ports:
+      - "8089:8089"
+    volumes:
+      - shared_root_dir:/
+    environment:
+      - DATASTORE_DATASET=dev-project-id
+      - DATASTORE_EMULATOR_HOST=0.0.0.0:8089
+      - DATASTORE_EMULATOR_HOST_PATH=0.0.0.0:8089/datastore
+      - DATASTORE_HOST=http://0.0.0.0:8089
+      - DATASTORE_PROJECT_ID=dev-project-id
+      - DATASTORE_USE_PROJECT_ID_AS_APP_ID=true
+      - GOOGLE_CLOUD_PROJECT=dev-project-id
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+      - oppia_is_dockerized=true
+    depends_on:
+      webserver:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8089/ || exit 1
+      timeout: 10s
+      interval: 2m
+      retries: 5
+      start_period: 20m
+    # Line 110 is checking if previous datastore cache exists and deleting it if it does as user has not passed the save_datastore flag.
+    # TODO: thouroughly test this.
+    command: >
+      sh -c '
+        if [ "$save_datastore" = "true" ]
+        then
+          /google-cloud-sdk/bin/gcloud beta emulators datastore start --project dev-project-id --data-dir /app/cloud_datastore_emulator_cache --host-port 0.0.0.0:8089 --consistency=1.0 --quiet
+        else
+          if [ -d /app/cloud_datastore_emulator_cache ]
+          then
+            rm -rf /app/cloud_datastore_emulator_cache
+          fi
+          /google-cloud-sdk/bin/gcloud beta emulators datastore start --project dev-project-id --data-dir /app/cloud_datastore_emulator_cache --host-port 0.0.0.0:8089 --consistency=1.0 --quiet --no-store-on-disk
+        fi
+      '
+
+  firebase:
+    image: node:16.13.0-alpine
+    container_name: oppia-firebase-emulator
+    ports:
+      - "9099:9099"
+      - "4000:4000"
+    volumes:
+      - shared_root_dir:/
+    healthcheck:
+      test: wget --spider http://localhost:4000/auth/ || exit 1
+      timeout: 10s
+      interval: 30s
+      retries: 5
+    environment:
+      - oppia_is_dockerized=true
+    command: >
+      sh -c '
+        if [ "$save_datastore" = "true" ]
+        then
+          /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js emulators:start --only auth --project dev-project-id --config /app/oppia/.firebase.json --import /app/firebase_emulator_cache --export-on-exit
+        else
+          /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js emulators:start --only auth --project dev-project-id --config /app/oppia/.firebase.json --export-on-exit /app/firebase_emulator_cache
+        fi
+      '
```

```suggestion
       firebase emulators:start \
       --only auth \
       --project=dev-project-id \
       --config=/app/oppia/.firebase.json \
       --export-on-exit \ 
       ${save_datastore:+--import}
       /app/firebase_emulator_cache 
```

Can we do something around this^?